include/picongpu/particles/ionization/byField/Keldysh/Keldysh_Impl.hpp

```diff
@@ -225,16 +221,59 @@ namespace ionization
                 /* define number of bound macro electrons before ionization */
                 float_X prevBoundElectrons = particle[boundElectrons_];
 
-                /* this is the point where actual ionization takes place */
                 IonizationAlgorithm ionizeAlgo;
-                ionizeAlgo(
+                /* determine number of new macro electrons to be created */
+                unsigned int newMacroElectrons = ionizeAlgo(
                      bField, eField,
                      particle, this->randomGen()
                      );
 
-                /* determine number of new macro electrons to be created */
-                newMacroElectrons = prevBoundElectrons - particle[boundElectrons_];
+                return newMacroElectrons;
+
+            }
+
+            /* Functor implementation
+             *
+             * Ionization model specific particle creation
+             *
+             * \tparam T_parentIon type of ion species that is being ionized
+             * \tparam T_childElectron type of electron species that is created
+             * \param parentIon ion instance that is ionized
+             * \param childElectron electron instance that is created
+             */
+            template<typename T_parentIon, typename T_childElectron>
+            DINLINE void operator()(T_parentIon& parentIon,T_childElectron& childElectron)
```

Comment:
Yeah, one could still keep it in the `ionizationMethods.hpp` and specify a different one for a special model.